examples/darwin-framework-tool/commands/pairing/OpenCommissioningWindowCommand.mm

```diff
@@ -17,47 +17,57 @@
 
 #import <Matter/Matter.h>
 
+#import "MTRError_Utils.h"
+
 #include "OpenCommissioningWindowCommand.h"
 
 CHIP_ERROR OpenCommissioningWindowCommand::RunCommand()
 {
+    mWorkQueue = dispatch_queue_create("com.chip.open_commissioning_window", DISPATCH_QUEUE_SERIAL);
     auto * controller = CurrentCommissioner();
-    NSError * error;
-    __block NSString * pairingCode;
+    auto * device = [[MTRBaseDevice alloc] initWithNodeID:@(mNodeId) controller:controller];
+
+    auto * self = this;
     if (mCommissioningWindowOption == 0) {
-        [controller openPairingWindow:mNodeId duration:mCommissioningWindowTimeoutMs error:&error];
+        auto * cluster = [[MTRBaseClusterAdministratorCommissioning alloc] initWithDevice:device endpoint:0 queue:mWorkQueue];
+        auto * params = [[MTRAdministratorCommissioningClusterOpenBasicCommissioningWindowParams alloc] init];
+        params.commissioningTimeout = @(mCommissioningWindowTimeoutMs);
+        params.timedInvokeTimeoutMs = @(10000);
```

Comment:
Should we add `--timedInteractionTimeoutMs` to the command (and make it a require argument, not an optional).

Reply:
The ECM version (which uses the dedicated API) would not use it, so making it required would be a bit odd.. and adding it at all would be odd-ish too, if it's ignored for the ECM version.